apps/public/src/app/(general)/samenvatting/page.tsx

```diff
@@ -21,64 +19,33 @@ export const generateMetadata = async () => {
 }
 
 export default async () => {
-  // Get session variables from cookies
   const cookieStore = await cookies()
   const meldingId = cookieStore.get('id')?.value
   const token = cookieStore.get('token')?.value
   const location = cookieStore.get('location')?.value
 
-  if (!meldingId || !token) return undefined
-
-  // Primary form
-
-  const { data: staticFormsData, error: staticFormsError } = await getStaticForm()
-
-  if (staticFormsError) return handleApiError(staticFormsError)
-
-  const primaryFormId = staticFormsData?.find((form) => form.type === 'primary')?.id
-
-  if (!primaryFormId) return 'Primary form id not found'
-
-  const { data: primaryFormData, error: primaryFormError } = await getStaticFormByStaticFormId({
-    path: { static_form_id: primaryFormId },
-  })
-
-  if (primaryFormError) return handleApiError(primaryFormError)
-
-  if (!primaryFormData) return 'Primary form data not found'
-
-  const primaryForm = primaryFormData.components[0]
-
-  // Melding
-
-  const { data: meldingData, error: meldingError } = await getMeldingByMeldingIdMelder({
-    path: { melding_id: parseInt(meldingId, 10) },
-    query: { token },
-  })
-
-  if (meldingError) return handleApiError(meldingError)
-
-  if (!meldingData) return 'Melding data not found'
+  const t = await getTranslations()
 
-  // Additional questions
+  if (!meldingId || !token) throw new Error('Could not retrieve meldingId or token')
 
-  const { data: additionalQuestionsData, error: additionalQuestionsErrors } = await getMeldingByMeldingIdAnswers({
-    path: { melding_id: parseInt(meldingId, 10) },
-    query: { token },
-  })
+  const meldingData = await getMeldingData(meldingId, token)
+  if (typeof meldingData === 'string') return meldingData
 
-  if (additionalQuestionsErrors) return handleApiError(additionalQuestionsErrors)
+  const additionalQuestionsAnswersSummary = await getAdditionalQuestionsSummary(meldingId, token)
+  if (typeof additionalQuestionsAnswersSummary === 'string') return additionalQuestionsAnswersSummary
 
-  const t = await getTranslations()
+  const primaryFormSummary = await getPrimaryFormSummary(meldingData.text)
+  if (typeof primaryFormSummary === 'string') return primaryFormSummary
 
-  const data = getSummaryData({
-    melding: meldingData,
-    primaryFormLabel: primaryForm?.label ?? '',
-    additionalQuestionsAnswers: additionalQuestionsData ?? [],
-    location: location ? JSON.parse(location) : undefined,
-    locationLabel: t('location.title'),
-    contactLabel: t('summary.contact-label'),
-  })
+  const locationSummary = getLocationSummary(t('location.title'), location)
+  const contactSummary = getContactSummary(t('summary.contact-label'), meldingData?.email, meldingData?.phone)
```

Comment:
These functions pretty much just return the object we need, maybe we don't really need to extract them into a function.

Reply:
Might be a bit overkill indeed. But they are there now.

Reply:
Well, they aren't really. You're proposing to add them. I'm proposing we don't 😁

Reply:
potato potato I'll remove it.